src/Forms/GridField/GridFieldDeleteAction.php

```diff
@@ -179,29 +179,51 @@ public function getColumnContent($gridField, $record, $columnName)
      */
     public function handleAction(GridField $gridField, $actionName, $arguments, $data)
     {
-        if ($actionName == 'deleterecord' || $actionName == 'unlinkrelation') {
+        $handledActions = $this->getActions($gridField);
+
+        if (in_array($actionName, $handledActions)) {
             /** @var DataObject $item */
             $item = $gridField->getList()->byID($arguments['RecordID']);
             if (!$item) {
                 return;
             }
 
-            if ($actionName == 'deleterecord') {
-                if (!$item->canDelete()) {
-                    throw new ValidationException(
-                        _t(__CLASS__ . '.DeletePermissionsFailure', "No delete permissions")
-                    );
-                }
-
-                $item->delete();
-            } else {
-                if (!$item->canEdit()) {
-                    throw new ValidationException(
-                        _t(__CLASS__ . '.EditPermissionsFailure', "No permission to unlink record")
+            switch ($actionName) {
+                case 'archiverecord':
+                    if (!$item->canArchive()) {
+                        throw new ValidationException(
+                            _t(__CLASS__ . '.ArchivePermissionsFailure', "No archive permissions")
+                        );
+                    }
+
+                    $item->doArchive();
+                    break;
```

Comment:
I'm don't think I agree with putting this code here. It's specific to the versioned module

Reply:
Perhaps we can put this into a versioned extension on this class instead?